src/Integration.Vsix/Suppression/DelegateInjector.cs

```diff
@@ -61,8 +71,37 @@ private void OnAssemblyLoad(object sender, AssemblyLoadEventArgs args)
 
         private void InjectSuppressionDelegate(Assembly asm)
         {
-            // TODO: if this is a Sonar analyzer assembly, try to set the suppression delegate
+            // If this is a Sonar analyzer assembly, try to set the suppression delegate
             // Note: the property might not exist for down-level versions of the analyzer
+            const string AssemblyName = "SonarAnalyzer";
+            const string NamespaceName = "SonarAnalyzer.Helpers";
+            const string ClassName = "SonarAnalysisContext";
+            const string PropertyName = "ShouldDiagnosticBeReported";
+
+            if (asm.FullName.StartsWith(AssemblyName, StringComparison.OrdinalIgnoreCase))
+            {
+                Type baseType = asm.GetTypes().SingleOrDefault<Type>(t =>
+                    t.Namespace.Equals(NamespaceName, StringComparison.Ordinal) &&
+                    t.Name.Equals(ClassName, StringComparison.Ordinal));
```

Comment:
Why not just something like `Type baseType = asm.GetType(Namespace + "." + ClassName)`?

Reply:
Done. Not sure what I was thinking :-)